addons/web/static/src/search/search_component.js

```diff
@@ -0,0 +1,13 @@
+/** @odoo-module **/
+
+import { SearchModel } from "./search_model";
+
+const { Component } = owl;
+
+export class SearchComponent extends Component {
+    setup() {
+        if (!(this.env.searchModel instanceof SearchModel)) {
+            throw new Error("A SearchModel instance in env is required");
```

Comment:
What about the client actions that have a control panel ? They don't have a searchModel.

Reply:
For now, the idea is to have one always but we can change that of course